logprep/connector/http/input.py

```diff
@@ -172,22 +172,40 @@ async def func_wrapper(*args, **kwargs):
     return func_wrapper
 
 
+default_meta_headers = frozenset(
+    [
+        "url",
+        "remote_addr",
+        "user-agent",
+    ]
+)
+
+
 def add_metadata(func: Callable):
     """Decorator to add metadata to resulting http event.
     Uses attribute collect_meta of endpoint class to decide over metadata collection
     Uses attribute metafield_name to define key name for metadata
     """
 
     async def func_wrapper(*args, **kwargs):
-        req = args[1]
-        endpoint = args[0]
+        req: falcon.Request = args[1]
+        endpoint: HttpEndpoint = args[0]
+
         if endpoint.collect_meta:
-            metadata = {
-                "url": req.url,
-                "remote_addr": req.remote_addr,
-                "user_agent": req.user_agent,
-            }
+            metadata = {}
+            for header in endpoint.copy_headers_to_logs:
+                # Remote_addr and url are special cases, because those are not copied 1 to 1 from headers
+                match header:
+                    case "remote_addr":
+                        metadata[header] = req.remote_addr
+                    case "url":
+                        metadata[header] = req.url
+                    case _:
+                        key = header.replace("-", "_").lower()
+                        metadata[key] = req.get_header(header, False)
```

Comment:
This line has been misleading to me (as someone who read the code on GitHub and is not familiar with falcon). Often, `get_xx` with `False` as a second positional parameters means that `False` is to be returned as a default value. Here it meant that the it's not *required* and thus no error should be raised if missing.

I would propose either

```suggestion
                        metadata[key] = req.get_header(header, default=None)
```

or 

```suggestion
                        metadata[key] = req.get_header(header, required=False, default=None)
```

Functionally both are equivalent to `metadata[key] = req.get_header(header)` but more readable IMO.